### PR TITLE
Players are comparable for sorting and implement Codable instead of N…

### PR DESCRIPTION
**Pull request by ChrisThirtle**
…SCoder

### CODE REVIEW

File: BoardBank/MonoBank/BankManager.swift [hunk 1]
```diff
@@ -98,6 +99,8 @@ final class BankManager {
   }
   
   func swapOrder(player: Player, otherPlayer: Player) {
-  //TODO: Implement swapping player order
+    print(player.index)
+    print(otherPlayer.index)
+    (player.index, otherPlayer.index) = (otherPlayer.index, player.index)
```
Comment (prrenfrew) [inline BoardBank/MonoBank/BankManager.swift]: Should you save the order after swapping them?

File: BoardBank/MonoBank/BankManager.swift [hunk 2]
```diff
@@ -98,6 +99,8 @@ final class BankManager {
   }
   
   func swapOrder(player: Player, otherPlayer: Player) {
-  //TODO: Implement swapping player order
+    print(player.index)
+    print(otherPlayer.index)
```
Comment (prrenfrew) [inline BoardBank/MonoBank/BankManager.swift]: Remove debug print statements.

File: BoardBank/MonoBank/MainViewController.swift [hunk 3]
```diff
@@ -227,8 +227,8 @@ class MainViewController: UIViewController, UICollectionViewDataSource, UICollec
   func collectionView(_ collectionView: UICollectionView, moveItemAt sourceIndexPath: IndexPath, to destinationIndexPath: IndexPath) {
     guard destinationIndexPath.item != 0 else { return }
     let players = bankManager.players
-    bankManager.swapOrder(player: players[sourceIndexPath.item],
-                          otherPlayer: players[destinationIndexPath.item])
+    bankManager.swapOrder(player: players[sourceIndexPath.item-1],
```
Comment (prrenfrew) [inline BoardBank/MonoBank/MainViewController.swift]: There should be a space on either side of `-`

File: BoardBank/MonoBank/Player+CoreDataClass.swift [hunk 4]
```diff
@@ -10,25 +10,13 @@
...
 
-//TODO: - remove NSCoding once we are using CoreData instead of UserDefaults
 @objc(Player)
-final public class Player: NSManagedObject, NSCoding {
-  // Methods to conform to NSCoding
-  
-  required convenience public init?(coder aDecoder: NSCoder) {
-    
-    //Mark: - Check if problems with Core Data
+final public class Player: NSManagedObject, Codable {
```
Comment (prrenfrew) [inline BoardBank/MonoBank/Player+CoreDataClass.swift]: Why does this have Codable?

File: BoardBank/MonoBank/Player+CoreDataProperties.swift [hunk 5]
```diff
@@ -30,3 +31,17 @@ extension Player {
...
+  public static func <= (lhs: Player, rhs: Player) -> Bool {
+    return lhs.index <= rhs.index
+  }
+  public static func > (lhs: Player, rhs: Player) -> Bool {
+    return lhs.index > rhs.index
+  }
+  public static func >= (lhs: Player, rhs: Player) -> Bool {
+    return lhs.index >= rhs.index
+  }
+}
```
Comment (prrenfrew) [inline BoardBank/MonoBank/Player+CoreDataProperties.swift]: Blank line at end of file.

File: BoardBank/MonoBank/Player+CoreDataProperties.swift [hunk 6]
```diff
@@ -30,3 +31,17 @@ extension Player {
   }
 }
 
+extension Player: Comparable {
+  public static func < (lhs: Player, rhs: Player) -> Bool {
```
Comment (prrenfrew) [inline BoardBank/MonoBank/Player+CoreDataProperties.swift]: You only need to implement `==` . and `<` for Comparable.

**Comment by prrenfrew** [comment]
@ChrisThirtle this has merge conflicts. Please fix

**Comment by fordhamite** [comment]
i agree

**Pull request merged**